### ISSUE TITLE
SetTotalRecords in CollectionEngine

### ISSUE THREAD

**Issue by Nks** [yajra/laravel-datatables#702]
### Summary of problem or feature request

Using collection we can't rewrite recordsTotal with ```setTotalRecords()``` method because it's not using in the CollectionEngine.php. 

### Code snippet of problem

Trying to do something like this:
```
return Datatables::of(new Collection($result))
                         ->setTotalRecords(100000)
                         ->make(true);
```
And it always return me only 10 (this is how much items we have in collection).
In [CollectionEngine.php](https://github.com/yajra/laravel-datatables/blob/6.0/src/Engines/CollectionEngine.php#L104) it calling method `count()` from Collection method. It seems we can add here ability to rewrite count. In this case we can get data from somewhere with limits and offset and rewrite totalRecords with it.

### System details

- Ubuntu 14.04.4 LTS
- PHP 7.0.3-13+deb.sury.org~trusty+1 (cli) ( NTS )
- Laravel 5.2.43
- Laravel-Datatables 6.17.0

**Comment by yajra** [comment]
Fix on v6.17.1, thanks for reporting!

**Comment by Nks** [comment]
@yajra many thanks for updates!
